src/DataShare/DataShare/Share/NewAzDataShare.cs

```diff
@@ -0,0 +1,111 @@
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Models;
+    using System.Management.Automation;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Extensions;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Properties;
+
+    /// <summary>
+    /// Defines the New-DataShare cmdlet.
+    /// </summary>
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataShare", SupportsShouldProcess = true, DefaultParameterSetName = ParameterSetNames.FieldsParameterSet), OutputType(typeof(PSDataShare))]
+    public class NewAzDataShare : AzureDataShareCmdletBase
+    {
+        /// <summary>
+        /// The resource group name of the azure data share account.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParameterSetNames.FieldsParameterSet,
+            HelpMessage = "The resource group name of the azure data share account")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Name of azure data share account.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParameterSetNames.FieldsParameterSet,
+            HelpMessage =  "Azure data share account name")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Account, "ResourceGroupName")]
+        public string AccountName { get; set; }
+
+        /// <summary>
+        /// Name of the azure data share.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParameterSetNames.FieldsParameterSet,
+            HelpMessage = "Azure data share name")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Share, "ResourceGroupName", "AccountName")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Description of the azure data share.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParameterSetNames.FieldsParameterSet,
+            HelpMessage = "Description of azure data share")]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// Terms of use of azure data share.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParameterSetNames.FieldsParameterSet, 
+            HelpMessage = "Terms of use for azure data share")]
+        public string TermsOfUse { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            this.ConfirmAction(
```

Comment:
Once again, you are combining ConfirmAction with ShouldProcess.  Please remove *all instances* of this